### PR TITLE
chore: Upgrade to Spring Boot 3.0.1

### LINKED ISSUE #18993: [Task]: Upgrade to Spring Boot 3
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [x] Upgrade to Spring Boot 3
- [x] Upgrade to JDK 17
- [ ] Upgrade Mongock libs
- [x] Upgrade vulnerable deps
- [x] Remove incompatible encryption migration

### PR DESCRIPTION
**Pull request by nidhi-nair**
This upgrade takes care of our move to JDK 17, Spring Boot 3.0.1 and a few other security upgrades along the way.

Fixes #18993 

TODO:

- [x] Check CI changes for Java 17
- [x] Check vulnerability report
- [x] Mongock needs an upgrade
- [x] Add JVM args at all possible places for exposing java.time module
- [x] Add type adapters everywhere / use the same config for type adapters everywhere

### CODE REVIEW

File: app/server/appsmith-server/src/main/resources/application.properties [hunk 1]
```diff
@@ -89,9 +91,9 @@ encrypt.password=${APPSMITH_ENCRYPTION_PASSWORD:}
 encrypt.salt=${APPSMITH_ENCRYPTION_SALT:}
 
 # The following configurations are to help support prometheus scraping for monitoring
-management.endpoints.web.exposure.include=prometheus
+management.endpoints.web.exposure.include=prometheus,metrics
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/resources/application.properties:94]: Don't need this

File: app/server/appsmith-server/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+mongock.change-logs-scan-package=com.appsmith.server.migrations
+# Ensures that the size of the request object that we handle is controlled. By default it's 212KB.
+spring.codec.max-in-memory-size=100MB
+appsmith.codec.max-in-memory-size=${APPSMITH_CODEC_SIZE:100}
+
+# MongoDB Application Database
+#spring.data.mongodb.uri = ${APPSMITH_MONGODB_URI}
+
+# embedded mongo DB version which is used during junit tests
+de.flapdoodle.mongodb.embedded.version=4.2.13
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/test/resources/application.properties:28]: Upgrade this

Comment (nidhi-nair) [reply app/server/appsmith-server/src/test/resources/application.properties:28]: Remove repetitions

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/helpers/restApiUtils/helpers/RestAPIActivateUtils.java [hunk 3]
```diff
@@ -78,7 +77,7 @@ public Mono<ActionExecutionResult> triggerApiCall(WebClient client, HttpMethod h
                         contentType = MediaType.TEXT_PLAIN;
                     }
                     byte[] body = stringResponseEntity.getBody();
-                    HttpStatus statusCode = stringResponseEntity.getStatusCode();
+                    var statusCode = stringResponseEntity.getStatusCode();
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/helpers/restApiUtils/helpers/RestAPIActivateUtils.java:80]: Change this

**Review by nidhi-nair — COMMENTED** [review]
Remove changes that are not relevant here.

**Review by sumitsum — APPROVED** [review]

**Pull request merged**